### PR TITLE
Update Task 7 & 8

### LINKED ISSUE #8: Task 7. (Player) New Class
Napisz klasę Player, która posiadać będzie 3 pola:

- std::shared_ptr<Ship> ship_ (dla chętnych, spróbuj użyć std::unique_ptr<>)
- money_
- availableSpace_

Dopisz także odpowiednie gettery oraz konstruktor.

Dopisz także 2 funkcje, które powinny zwracać dane ze statku:

- size_t getSpeed() const
- Cargo* getCargo(size_t index) const

### LINKED ISSUE #9: Task 8. (Player) availableSpace
W klasie Player napisz prywatną funkcję, która obliczać będzie availableSpace_ na podstawie aktualnej ilości towaru na statku.

### PR DESCRIPTION
**Pull request by Pawellos**
Jeszcze coś nie kompiluje, powalczę z tym ale trochę poprawiłem.

### CODE REVIEW

File: shm/player.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include "player.hpp"
+
+#include <numeric>
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
```
Comment (g00fer) [inline shm/player.cpp]: Please do not use size_t. Explanation on squad-7 discord channel.

File: shm/player.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace) {}
+
+std::unique_ptr<Ship> Player::getShip() {
+    return std::move(ship_);
```
Comment (g00fer) [inline shm/player.cpp:9]: Are you absolutely sure about that operation? We are not at this level in this course :)

**Review by g00fer — CHANGES_REQUESTED** [review]
please replace all size_t with proper types

File: shm/player.cpp (hunk 1 above)
Comment (Pawellos) [reply shm/player.cpp]: I replaced all size_t on uint_32t, now is good ?

File: shm/player.cpp (hunk 2 above)
Comment (Pawellos) [reply shm/player.cpp:9]: Maybe not absolutely but i am pretty sure about that operation. In task was write:
Napisz klasę Player, która posiadać będzie 3 pola:
    std::shared_ptr<Ship> ship_ (dla chętnych, spróbuj użyć std::unique_ptr<>)
So if i am using unique_ptr i am not see any other solution.

File: shm/ship.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include "ship.hpp"
+
+#include <iostream>
+#include <string>
+
+Ship::Ship() : id_(-1){};
+
+Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
```
Comment (Kasperice) [inline shm/ship.cpp]: replace all int't to uint32_t. Maybe in this class it is not necessary, but you can do that, there will be less conflicts during merge ;)

File: shm/ship.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id){};
+
+Ship::Ship(int maxCrew, int speed, size_t id) : Ship(0, maxCrew, speed, "", id){};
+
+void Ship::setName(const std::string& name) {
+    name_ = name;
+}
+
+Ship& Ship::operator-=(size_t crew) {
```
Comment (Kasperice) [inline shm/ship.cpp]: size_t --> uint32_t

File: shm/ship.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Ship& Ship::operator+=(size_t crew) {
+    if (crew_ + crew <= maxCrew_) {
+        crew_ += crew;
+        return *this;
+    }
+    std::cerr << "Too many sailors!";
+    return *this;
+}
+
+size_t Ship::getId() const {
```
Comment (Kasperice) [inline shm/ship.cpp]: and all of those size_t --> uint32_t

**Review by Kasperice — CHANGES_REQUESTED** [review]

File: shm/ship.hpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    Cargo* getCargo(size_t index);
+    std::vector<Cargo> getVectorCargo() const;
+
+private:
+    size_t id_;
+    std::string name_;
+    size_t speed_;
+    size_t maxCrew_;
+    size_t capacity_;
+    size_t crew_;
```
Comment (Kasperice) [inline shm/ship.hpp]: Here also size_t, int --> uint32_t

**Review by Kasperice — CHANGES_REQUESTED** [review]

**Comment by Kasperice** [comment]
@Pawellos did you check if it is compiling now?

**Review by Kasperice — APPROVED** [review]

**Pull request merged**